### DISCUSSION TITLE
Agnostic NMS when loading from PyTorch Hub

### DISCUSSION THREAD

**Discussion by c-axel** [ultralytics/yolov5#2370] [Q&A] [answered]
Is there a way to use agnostic NMS when loading a custom model from PyTorch Hub? Trying to replicate what I get when using detect.py with --agnostic. 

I can copy the approach in utils.general.non_max_suppression(agnostic=True) in my script and apply it on each result, but curious if there is a more elegant way.

**Comment by glenn-jocher** [comment, marked as answer]
@c-axel this is not available currently, but if you'd like to submit a PR you can introduce this in the autoShape() class:
https://github.com/ultralytics/yolov5/blob/e931b9da33f45551928059b8d61bddd50e401e48/models/common.py#L168-L229

**Reply by c-axel** [reply]
Understood, thanks for the quick reply and thank you for yolov5 it is awesome.

**Reply by glenn-jocher** [reply]
Thanks, we've tried to make it better by listening to feedback and implementing many small ideas just like this one!
